scripts/config.rb

```diff
@@ -0,0 +1,15 @@
+require 'pathname'
+
+BUILD_BASE          = Pathname.new("/build")
+GPG_DIR             = Pathname.new("/root/.gnupg")
+CFG_DIR             = BUILD_BASE.join("config")
+CREDS_DIR           = CFG_DIR.join("creds")
+FILESHARE_DIR       = BUILD_BASE.join("fileshare")
+REFS_DIR            = BUILD_BASE.join("references")
+IMGFAC_DIR          = BUILD_BASE.join("imagefactory")
+IMGFAC_CONF         = CFG_DIR.join("imagefactory.conf")
+STORAGE_DIR         = BUILD_BASE.join("storage")
+
+FILE_SERVER         = ENV["BUILD_FILE_SERVER"]             # SSH Server to host files
+FILE_SERVER_ACCOUNT = ENV["BUILD_FILE_SERVER_ACCOUNT"]     # Account to SSH as
+FILE_SERVER_BASE    = Pathname.new(ENV["BUILD_FILE_SERVER_BASE"] || ".") # Subdirectory of Account where to store builds
```

Comment:
So, we need to put these constants within the `Build` namespace.  Meaning, they should reside in `module Build`.  We can do this later but we really should have them in that namespace and all code that uses it, if it's in that namespace, can access it without needing to specify the namespace, meaning we can use `BUILD_BASE` within the namespace instead of  `Build::BUILD_BASE`.  When we put it in the `Build` namespace, it needs to live in a build directory, so `scripts/build/config.rb`...

Because it's relevant here but not required... I created #158 for the following comment:

Future PR: Note, we really need to move files to the right places and re-namespace things...

```
vmbuild.rb             -> build.rb
cli.rb                 -> build/cli.rb
git_checkout.rb        -> build/git_checkout.rb
kickstart_generator.rb -> build/kickstart_generator.rb
productization.rb      -> build/productization.rb
target.rb              -> build/target.rb
productization.rb      -> build/productization.rb
uploader.rb            -> build/uploader.rb
```

Also, rename scripts to lib.